Automation/Task4/deploy_template_task4.ps1

```diff
@@ -0,0 +1,58 @@
+#parameters required to run the script.
+param(
+    [Parameter(Mandatory=$True)]
+    [string]
+    $SubscriptionId,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $TenantId,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ServicePrincipalId,
+
+    [Parameter(Mandatory=$True)]
+    [String]
+    $ServicePrincipalPassword,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $Resource_Group_Name,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $function_name,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $storage_account_name,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $app_service_plan_name,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $location,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $template_file_path
+
+
+
+)
+
+
+Clear-AzContext -Force;
+$securePassword = ConvertTo-SecureString -String $ServicePrincipalPassword -AsPlainText -Force;
+$credentials = New-Object -TypeName System.Management.Automation.PSCredential($ServicePrincipalId, $securePassword);
+
+
+Connect-AzAccount -Credential $credentials -ServicePrincipal -Tenant $TenantId -SubscriptionId $SubscriptionId;
+
+New-AzResourceGroup -Name $Resource_Group_Name -Location $location
+
+New-AzResourceGroupDeployment -ResourceGroupName $Resource_Group_Name -TemplateFile "./template.json" -function_name $function_name -storage_account_name $storage_account_name -app_service_plan_name $app_service_plan_name -location $location
```

Comment:
You can improve this by collecting the parameters in a HashTable. And you can pass the HashTable as parameter to New-AzResourceGtoupDeployment

Reply:
I have so far been unable to fix the script with the hashtable. However, i have fixed the naming consistency issue. If you are ok, then we can merge this one and keep the hashtable as a future improvement in one of the next sprints.